aten/src/TH/generic/THStorage.hpp

```diff
@@ -4,13 +4,29 @@
 
 typedef struct THStorage
 {
-    real *data;
+    at::ScalarType scalar_type;
+    void *data_ptr;
     ptrdiff_t size;
     std::atomic<int> refcount;
     char flag;
     THAllocator *allocator;
     void *allocatorContext;
     struct THStorage *view;
+    template <typename T>
+
+    inline T * data() const {
+      auto scalar_type_T = at::CTypeToScalarType<th::from_type<T>>::to();
+      if (scalar_type != scalar_type_T) {
+        AT_ERROR("Attempt to access Storage having data type ", at::toString(scalar_type),
+                 " as data type ", at::toString(scalar_type_T));
+      }
+      return unsafeData<T>();
+    }
+
+    template <typename T>
+    inline T * unsafeData() const {
```

Comment:
Prefer camel_case for function and variable names.

Reply:
I use you meant not camel_case?